// angular-svg/esm2015/lib/types/stroke.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
export class Stroke {
    /**
     * @param {?=} fill
     * @param {?=} strokeFill
     * @param {?=} dashArray
     * @param {?=} thickness
     * @param {?=} opacity
     */
    constructor(fill, strokeFill, dashArray, thickness, opacity) {
        this.fill = fill != null ? fill : 'transparent';
        this.strokeFill = strokeFill != null ? strokeFill : 'black';
        this.dashArray = dashArray != null ? dashArray : 0;
        this.thickness = thickness != null ? thickness : 1;
        this.opacity = opacity != null ? opacity : 1;
    }
}
if (false) {
    /** @type {?} */
    Stroke.prototype.fill;
    /** @type {?} */
    Stroke.prototype.strokeFill;
    /** @type {?} */
    Stroke.prototype.dashArray;
    /** @type {?} */
    Stroke.prototype.stroke_linecap;
    /** @type {?} */
    Stroke.prototype.thickness;
    /** @type {?} */
    Stroke.prototype.opacity;
}
