// angular-svg/esm5/angular-svg.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */
export { SvgCircleModule, SvgEllipseModule, SvgLineModule, SvgPathModule, SvgPolygonModule, SvgPolylineModule, SvgrectModule, SvgTextModule, Polygon, Polyline, Stroke, Vertex } from './public_api';
export { SvgHelper as ɵf } from './lib/helpers/svghelper.service';
export { SvgCircleComponent as ɵa } from './lib/modules/svgcircle/svgcircle.component';
export { SvgEllipseComponent as ɵb } from './lib/modules/svgellipse/svgellipse.component';
export { SvgLineComponent as ɵc } from './lib/modules/svgline/svgline.component';
export { SvgPathComponent as ɵd } from './lib/modules/svgpath/svgpath.component';
export { SvgPolygonComponent as ɵe } from './lib/modules/svgpolygon/svgpolygon.component';
export { SvgPolylineComponent as ɵg } from './lib/modules/svgpolyline/svgpolyline.component';
export { SvgRectComponent as ɵh } from './lib/modules/svgrect/svgrect.component';
export { SvgTextComponent as ɵi } from './lib/modules/svgtext/svgtext.component';
