// angular-svg/esm5/public_api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*
 * Public API Surface of angularsvg
 */
export { SvgCircleModule } from './lib/modules/svgcircle/svgcircle.module';
export { SvgEllipseModule } from './lib/modules/svgellipse/svgellipse.module';
export { SvgLineModule } from './lib/modules/svgline/svgline.module';
export { SvgPathModule } from './lib/modules/svgpath/svgpath.module';
export { SvgPolygonModule } from './lib/modules/svgpolygon/svgpolygon.module';
export { SvgPolylineModule } from './lib/modules/svgpolyline/svgpolyline.module';
export { SvgrectModule } from './lib/modules/svgrect/svgrect.module';
export { SvgTextModule } from './lib/modules/svgtext/svgtext.module';
export {} from './lib/types/circle.model';
export {} from './lib/types/ellipse.model';
export {} from './lib/types/path.model';
export { Polygon } from './lib/types/polygon.model';
export {} from './lib/types/elementclick-event.model';
export {} from './lib/types/line.model';
export { Polyline } from './lib/types/polyline.model';
export {} from './lib/types/rectangle.model';
export { Stroke } from './lib/types/stroke.model';
export {} from './lib/types/textfont.model';
export { Vertex } from './lib/types/vertex.model';
